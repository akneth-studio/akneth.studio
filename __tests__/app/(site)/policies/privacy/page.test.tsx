/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import PrivacyPage from '@/app/(site)/policies/privacy/page';

// Mock environment variables
process.env.NEXT_PUBLIC_SITE_URL = require('../../../../../jest.setup').NEXT_PUBLIC_SITE_URL || 'https://akneth.pl';

// Mock the PolicyView component since it's a server component
jest.mock('@/components/policies/policies', () => {
  return {
    __esModule: true,
    default: jest.fn(() => {
      return React.createElement('div', { 'data-testid': 'policy-view', 'data-filename': 'privacy.md' }, 'Policy Content for\nprivacy.md');
    })
  };
});

describe('PrivacyPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render PolicyView with correct filename', () => {
    render(<PrivacyPage />);
    
    // Check if PolicyView is rendered with correct filename
    const policyView = screen.getByTestId('policy-view');
    expect(policyView).toBeInTheDocument();
    expect(policyView).toHaveAttribute('data-filename', 'privacy.md');
    expect(policyView).toHaveTextContent('Policy Content for privacy.md');
  });

  it('should have correct metadata', () => {
    // Check if metadata is defined
    const { metadata } = require('@/app/(site)/policies/privacy/page');
    expect(metadata).toBeDefined();
    expect(metadata.title).toBe('Polityka prywatności');
    expect(metadata.description).toBe('Polityka prywatności AKNETH Studio Katarzyna Pawłowska-Malesa');
    expect(metadata.keywords).toEqual([
      'Polityka prywatności',
      'Prywatność',
      'Przetwarzanie danych',
      'Cookies',
      'Ciasteczka',
      'Bezpieczeństwo danych'
    ]);
    expect(metadata.alternates?.canonical).toBe('https://akneth.pl/policies/privacy');
  });

  it('should match snapshot', () => {
    const { container } = render(<PrivacyPage />);
    expect(container).toMatchSnapshot();
  });
});
