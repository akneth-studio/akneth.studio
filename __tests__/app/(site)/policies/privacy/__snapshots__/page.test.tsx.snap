// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PrivacyPage should match snapshot 1`] = `
<div>
  <div
    data-filename="privacy.md"
    data-testid="policy-view"
  >
    Policy Content for
privacy.md
  </div>
</div>
`;
